pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -484,6 +486,13 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 		return reconcile.Result{}, nil
 	}
 
+	// delete failed provisions which are more than 7 days old
+	existingProvisions, err := r.existingProvisions(cd, cdLog)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	r.deleteOldFailedProvisions(existingProvisions, cdLog)
```

Comment:
Do this in the `if cd.Spec.Installed` block. There is already code to clean up ClusterProvisions as new ClusterProvisions are created.